### DISCUSSION TITLE
Add folder from front end user

### DISCUSSION THREAD

**Discussion by GeorgeGeorgeGe** [rejetto/hfs#223] [General]
When we add a folder to the root folder from user browser using home>>upload >> Create Folder it is added successfully.
but if we go to the admin panel we will not find that new folder under the root folder ( inconsistency) and we need to add manually, now if we add it manually from there and delete it later from there as well, it will not deleted from the front end browser !!! and you need to re-add it again from the admin portal and delete it later from the front end user and then delete it from admin portal, which is a behaviour we do not experience when we add the folder from admin portal in the first place.
Please check.

**Comment by rejetto** [comment]
That's expected: "create folder" creates on disk.
What you see in the admin panel is the VIRTUAL file system. In the virtual file system you design your content and is separated from the disk. BUT you can set the "source" field of each folder/file, so that it takes the content from the disk.
You are not supposed to see in the VFS what's on disk.
If you have folder X, and it has a source, it will load stuff from disk. Such stuff will be loaded because you set the source.
Is it clearer now?

This may be not super-easy concept to grasp, i know, but if you have suggestions on how to make it clearer, I can consider it.


I'm considering a smarter visualization of the VFS, that will somehow show you also the content of the disk,  but it takes a lot of work and I have to think it through.

**Reply by GeorgeGeorgeGe** [reply]
You are 100% CORRECT !
in my scenario i made the home folder points to a physical folder on disk to allow front-end "admin-like" user to add folders to it when they need without the intervention of the Admin Panel administrator and that creates the confusion.
Thanks.
